app/jobs/scheduled/reindex_search.rb

```diff
@@ -66,6 +66,18 @@ def clean_post_search_data
         .joins("LEFT JOIN posts p ON p.id = post_search_data.post_id")
         .where("p.raw = ''")
         .delete_all
+
+      DB.exec(<<~SQL)
+        DELETE FROM post_search_data
+        WHERE post_id IN (
+          SELECT post_id
+          FROM post_search_data
+          LEFT JOIN posts ON post_search_data.post_id = posts.id
+          LEFT JOIN topics ON posts.topic_id = topics.id
+          WHERE topics.id IS NOT NULL
```

Comment:
Why not use an `INNER JOIN` if you require a topic?

Reply:
Ah good catch :+1: I've been very used to just doing a LEFT JOIN and checking for `NOT NULL`.